packages/react-server/src/ReactFizzServer.js

```diff
@@ -1567,17 +1600,26 @@ function renderChildrenArray(
   request: Request,
   task: Task,
   children: Array<any>,
+  childIndex: number,
 ) {
+  const prevKeyPath = task.keyPath;
   const totalChildren = children.length;
   for (let i = 0; i < totalChildren; i++) {
     const prevTreeContext = task.treeContext;
     task.treeContext = pushTreeContext(prevTreeContext, totalChildren, i);
     try {
+      const node = children[i];
+      if (isArray(node) || getIteratorFn(node)) {
+        // Nested arrays behave like a "fragment node" which is keyed.
+        // Therefore we need to add the current index as a parent key.
+        task.keyPath = [task.keyPath, '', childIndex];
```

Comment:
This is subtle but the first array inside of a component is the same as having a single element with slot `0`. So the array itself doesn't take up slot `0`. In other words, we the outer most array doesn't get added to the path but the next one gets treated as if it was a fragment element.